db/iscn.go

```diff
@@ -0,0 +1,159 @@
+package db
+
+import (
+	"encoding/json"
+	"fmt"
+	"log"
+	"strings"
+
+	"github.com/cosmos/cosmos-sdk/types"
+	"github.com/jackc/pgtype"
+	"github.com/jackc/pgx/v4"
+	"github.com/jackc/pgx/v4/pgxpool"
+	iscnTypes "github.com/likecoin/likechain/x/iscn/types"
+)
+
+type ISCNResponse struct {
+	iscnTypes.IscnRecord
+	Id              string `json:"@id"`
+	Type            string `json:"@type"`
+	RecordTimestamp string `json:"recordTimestamp"`
+	Owner           string `json:"owner"`
+}
+
+func QueryISCN(conn *pgxpool.Conn, events types.StringEvents, query ISCNRecordQuery, keywords Keywords, pagination Pagination) ([]iscnTypes.QueryResponseRecord, error) {
+	eventStrings := getEventStrings(events)
+	queryString, err := query.Marshal()
+	if err != nil {
+		return nil, err
+	}
+	keywordString := keywords.Marshal()
+	ctx, cancel := GetTimeoutContext()
+	defer cancel()
+
+	tx, err := conn.Begin(ctx)
+	if err != nil {
+		return nil, err
+	}
+	defer tx.Rollback(ctx)
+
+	// For GIN work with pagination
+	if _, err := tx.Exec(ctx, `SET LOCAL enable_indexscan = off;`); err != nil {
```

Comment:
Also would this affect `tx` query when concurrently called?

Reply:
> Can you explain a bit how does disabling index scan work for pagination?

The GIN index won't work with pagination for the default setting. According to [this post](https://dba.stackexchange.com/questions/113132/gin-index-not-used-when-adding-order-clause), this setting can fix the problem (and it works). 

> Also would this affect tx query when concurrently called?

It would affect other query so I wrap it into a transaction. With the LOCAL setting, it will only affect current transaction. I have tested it and it won't affect other queries now.